### PR TITLE
Clinical Timeline View

### PR DESCRIPTION
**Pull request by robertdown**
Related to #808

### CODE REVIEW

File: interface/globals.php [hunk 1]
```diff
@@ -234,6 +236,9 @@
 $twigEnv->addFilter(new Twig_SimpleFilter('translate', function ($string) {
     return xl($string);
 }));
+$twigEnv->addFunction(new Twig_SimpleFunction('setupHeader', function ($assets = []) {
+    return Header::setupHeader($assets);
```
Comment (bradymiller) [inline interface/globals.php]: Note this function will echo everything, so the return seems odd.

File: interface/globals.php [hunk 2]
```diff
@@ -4,6 +4,7 @@
 require_once(dirname(__FILE__) . "/../common/compatibility/Checker.php");
 
 use OpenEMR\Checker;
+use OpenEMR\Core\Header;
```
Comment (bradymiller) [inline interface/globals.php]: Doesn't this basically mean that we will not need to use the OpenEMR\Core\Header in any of the downstream scripts?

Comment (bradymiller) [inline interface/globals.php]: And I am assuming it's ok to use this even before the autoloader has been run?

File: interface/patient_file/summary/demographics.php [hunk 3]
```diff
@@ -622,6 +622,10 @@ function setMyPatient() {
           |
           <a href="../../reports/external_data.php" onclick='top.restoreSession()'>
           <?php echo xlt('External Data'); ?></a>
+          |
+          <a href="timeline.php?action=list" onclick="top.restoreSession()">
+              <?php echo xlt('Timeline'); ?>
+          </a>
```
Comment (bradymiller) [inline interface/patient_file/summary/demographics.php]: This is important enough to place it to the far left on this menu :)

File: interface/patient_file/summary/timeline.php [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+require_once "{$srcdir}/acl.inc";
+use OpenEMR\Patient\Timeline;
+use Symfony\Component\HttpFoundation\Request;
+
+if (!acl_check('patients', 'med')) {
+    die('Not authorized');
+}
+
+$request = Request::createFromGlobals();
+$action = $request->get('action');
```
Comment (bradymiller) [inline interface/patient_file/summary/timeline.php]: neat.
So this is basically a controller, right?

**Comment by bradymiller** [comment]
Regarding forms, ensure also grab the layout based forms (you may already be doing this). Also, I just realized that Rod recently added code which basically assigns an aco to each form (so shouldn't use the patients-med); check out the Administration->Other->Forms to see what I mean. And can look at the Patient->Reports for an example of how all these forms are collected.

File: interface/globals.php (hunk 2 above)
Comment (bradymiller) [inline interface/globals.php]: I am guessing in the future the plan is to migrate twig out of global.php into it's own place that is only called when using a twig template. If this is the case, just rec. placing a comment with this use OpenEMR\Core\Header (or could break the PSR rules and bundle it with the twig stuff below) so we remember to remove it when that happens.

File: templates/patient/timeline.html.twig [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+                <h1><i class="fa fa-history fa-lg"></i>&nbsp;{{ 'Clinical Timeline'|translate }}</h1>
+            </div>
+        </div>
+    </div>
+    <div class="row">
+        <div class="col-xs-4">
+            <div class="list-group">
+                {% for form in forms %}
+                <a href="timeline.php?action=detail&form_id={{ form.form_id|e('html_attr') }}" class="list-group-item {% if form.deleted %}text-muted{% endif %}">
+                    <span class="monospaced small">{{ form.date|date("Y-m-d") }}</span>
```
Comment (bradymiller) [inline templates/patient/timeline.html.twig]: Will need to have a twig wrapper for:
https://github.com/openemr/openemr/blob/master/library/formatting.inc.php#L20
(this converts YYYYMMDD to the date locale that is chosen in globals)

File: interface/globals.php (hunk 2 above)
Comment (robertdown) [inline interface/globals.php]: Maybe.... probably. But IDE integration will probably lack w/o that statement. And yes, we definitely want to break out the twig stuff (Don't worry, I'm definitely working on it, most of the puzzle pieces are in place :D )

File: interface/patient_file/summary/timeline.php (hunk 4 above)
Comment (robertdown) [inline interface/patient_file/summary/timeline.php]: Indeed!

**Comment by dimitristsilis** [comment]
Where do we access Timeline from in OpenEMR?

**Pull request closed without merging**